src/middlewares/product.middleware.js

```diff
@@ -0,0 +1,52 @@
+import { Products } from '../database/models/index';
+import { Categories } from '../database/models/index';
+import { uploadPhoto } from '../utils/cloudinary.util';
+
+export const checkIfProductExists = async (req, res, next) => {
+  const name = req.body.name;
+  const getProduct = await Products.findOne({
+    where: { name: name },
+  });
+  if (getProduct) res.status(404).json({ message: 'Product already exists' });
+  else next();
+};
+
+export const allowedExtensions = (...extensions) => {
+  return async (req, res, next) => {
+
+    const images = [];
+    for (let i = 0; i < req.files.images.length; i++) {
+      const ext = req.files.images[i].name.split('.').pop().toLowerCase();
+      if (extensions.includes(ext)) {
+        const { url } = await uploadPhoto(req, res, req.files.images[i]);
+        images.push(url);
+      }
+    }
+    req.images = images;
+    next();
+  };
+};
```

Comment:
what happens when the extension is invalid?  try to add an else condition to throw an error